packages/experimental/Checkbox/src/Checkbox.tsx

```diff
@@ -0,0 +1,41 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { View } from 'react-native';
+import { checkboxName, CheckboxType, CheckboxProps } from './Checkbox.types';
+import { Text } from '@fluentui-react-native/experimental-text';
+import { stylingSettings } from './Checkbox.styling';
+import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
+import { useCheckbox } from './useCheckbox';
+
+export const Checkbox = compose<CheckboxType>({
+  displayName: checkboxName,
+  ...stylingSettings,
+  slots: {
+    root: View,
+    checkbox: View,
+    checkmark: Text,
+    content: Text,
+  },
+  render: (userProps: CheckboxProps, useSlots: UseSlots<CheckboxType>) => {
+    const Checkbox = useCheckbox(userProps);
+    // grab the styled slots
+    const Slots = useSlots(userProps, layer => Checkbox.state[layer] || userProps[layer]);
+    // now return the handler for finishing render
+    return (final: CheckboxProps, ...children: React.ReactNode[]) => {
+      const { label, ...mergedProps } = mergeProps(Checkbox.props, final);
+
+      return (
+        <Slots.root {...mergedProps}>
+          {Checkbox.state.boxAtEnd && <Slots.content key="content">{label}</Slots.content>}
+          <Slots.checkbox>
+            <Slots.checkmark key="checkmark">✓</Slots.checkmark>
```

Comment:
nit: It's a bit unfortunate that we do "unchecked" by rendering the checkmark with 0 opacity, since that's more costly from a rendering perspective than not rendering the checkmark. It would only be a small perf optimization though so not a blocker.

Reply:
It just occurred to me that if we only render the checkmark when it's checked (instead of rendering it with 0 vs 1 opacity) then we can't have a checkbox where the checkmark is visible on hover (which is one of our examples in the Fluent Tester). For this reason, I may change the behavior back to what it was since I can't see how else someone would achieve that without messing with the opacity of an already rendered checkmark.

Reply:
Yeah that's a good catch. Makes sense to me.